go-controller/pkg/cluster/ha_master.go

```diff
@@ -0,0 +1,541 @@
+		// the cache. It is better to exit for now.
+		// kube will restart and this will become a follower.
+		if hacluster.manageDBServers {
+			// Stop ovn-northd and then exit.
+			_, _, _ = util.RunOVNNorthAppCtl("exit")
+		}
+		logrus.Infof("I (" + hacluster.nodeName + ") am no longer a leader. Exiting")
+		os.Exit(1)
+	}
+
+	HAClusterNewLeader := func(nodeName string) {
+		logrus.Infof(nodeName + " is the new leader")
+		wasLeader := hacluster.isLeader
+
+		if hacluster.nodeName == nodeName {
+			// Configure as leader.
+			logrus.Infof(" I (" + hacluster.nodeName + ") won the election. In active mode")
+			err = hacluster.ConfigureAsActive(nodeName)
+			if err != nil {
+				logrus.Errorf(err.Error())
+				if hacluster.manageDBServers {
+					// Stop ovn-northd before panicing.
+					_, _, _ = util.RunOVNNorthAppCtl("exit")
+				}
+				panic(err.Error())
+			}
+			hacluster.isLeader = true
+		} else if wasLeader {
+			HAClusterOnStoppedLeading()
+			// should not be reached
+			panic("This should not happen.")
+		} else {
+			// Configure as standby.
+			logrus.Infof(" I (" + hacluster.nodeName + ") lost the election. In Standby mode")
+			ep, er := hacluster.cluster.Kube.GetEndpoint(config.Kubernetes.OVNConfigNamespace, ovnkubeDbEp)
+			if er == nil {
+				er = hacluster.ConfigureAsStandby(ep)
+				if er != nil {
+					logrus.Errorf(er.Error())
+					if hacluster.manageDBServers {
+						// Stop ovn-northd and then exit
+						_, _, _ = util.RunOVNNorthAppCtl("exit")
+					}
+					panic(er.Error())
+				}
+			}
+		}
+	}
+
+	lec := leaderelection.LeaderElectionConfig{
+		Lock:          rl,
+		LeaseDuration: leaseDuration,
+		RenewDeadline: renewDeadline,
+		RetryPeriod:   retryPeriod,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: func(ctx context.Context) {},
+			OnStoppedLeading: HAClusterOnStoppedLeading,
+			OnNewLeader:      HAClusterNewLeader,
+		},
+	}
+
+	//go leaderelection.RunOrDie(context.Background(), lec)
+	hacluster.leaderElector, err = leaderelection.NewLeaderElector(lec)
+	if err != nil {
+		return err
+	}
+
+	go hacluster.leaderElector.Run(context.Background())
+
+	return hacluster.WatchOvnDbEndpoints()
+}
+
+// ConfigureAsActive configures the node as active.
+func (hacluster *OvnHAMasterController) ConfigureAsActive(masterNodeName string) error {
+	if hacluster.manageDBServers {
+		// Step 1: Update the ovnkube-db endpoints with invalid Ip.
+		// Step 2: Promote OVN DB servers to become active
+		// Step 3: Make sure that ovn-northd has done one round of
+		//         flow computation.
+		// Step 4: Update the ovnkube-db endpoints with the new master Ip.
+
+		// Find the endpoint for the service
+		ep, err := hacluster.cluster.Kube.GetEndpoint(config.Kubernetes.OVNConfigNamespace, ovnkubeDbEp)
+		if err != nil {
+			ep = nil
+		}
+		err = hacluster.updateOvnDbEndpoints(ep, true)
+		if err != nil {
+			logrus.Errorf("%s Endpount create/update failed", ovnkubeDbEp)
```

Comment:
Endpount -> Endpoint

Reply:
@numansiddique there's a couple more instances of Endpount to fix...

Reply:
Damn me.
I will fix it.

Reply:
Dobe